Joda Recipe NullpointerException when visting Java Records

<!--
Thank you for reporting an issue with OpenRewrite!
We appreciate you taking the time to help us improve.
Please fill out the template below to help us understand and reproduce the issue.
Feel free to delete any sections that don't apply to your issue.
-->

## What version of OpenRewrite are you using?
<!--
Whenever possible please try to replicate your issue with the latest versions of OpenRewrite.
The latest major and minor versions of OpenRewrite projects are usually listed here:
https://docs.openrewrite.org/reference/latest-versions-of-every-openrewrite-module
For patch releases check the GitHub Releases page for the respective project.

We release every few weeks, so it's possible that your issue has already been fixed.

If you want to try the most recent changes that haven't been fully released yet, you can check out our snapshot releases
https://docs.openrewrite.org/reference/snapshot-instructions
-->
I am using

- Java 21
- Maven CLI 3.9.6
- rewrite-migrate-java-2.31.0

## How are you running OpenRewrite?
<!--
Are you using the Maven plugin, Gradle plugin, Moderne CLI, Moderne.io or something else?
Is your project a single module or a multi-module project?

Can you share your configuration so that we can rule out any configuration issues?

Is your project public? If so, can you share a link to it?
Code snippets can also be shared privately via [our public Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA).
-->
I am using the Maven CLI , and my project is a multi module project.
mvn -U org.openrewrite.maven:rewrite-maven-plugin:run -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-migrate-java:RELEASE -Drewrite.activeRecipes=org.openrewrite.java.migrate.joda.JodaTimeRecipe -Drewrite.exportDatatables=true


## What is the full stack trace of any errors you encountered?
<!-- When errors occur, please include the output of `--stacktrace` for Gradle or `--debug` for Maven. -->
```
java.lang.NullPointerException: Cannot read field "variables" because "x0" is null
[ERROR]   org.openrewrite.java.migrate.joda.ScopeAwareVisitor$VariablesInScope.access$000(ScopeAwareVisitor.java:86)
[ERROR]   org.openrewrite.java.migrate.joda.ScopeAwareVisitor.preVisit(ScopeAwareVisitor.java:46)
[ERROR]   org.openrewrite.java.migrate.joda.ScopeAwareVisitor.preVisit(ScopeAwareVisitor.java:31)
[ERROR]   org.openrewrite.TreeVisitor.visit(TreeVisitor.java:247)
[ERROR]   org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:320)
[ERROR]   org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1367)
[ERROR]   org.openrewrite.java.JavaVisitor.lambda$visitVariableDeclarations$27(JavaVisitor.java:961)
[ERROR]   org.openrewrite.internal.ListUtils.map(ListUtils.java:243)
[ERROR]   org.openrewrite.internal.ListUtils.map(ListUtils.java:265)
[ERROR]   org.openrewrite.java.JavaVisitor.visitVariableDeclarations(JavaVisitor.java:961)
[ERROR]   org.openrewrite.java.tree.J$VariableDeclarations.acceptJava(J.java:5812)
[ERROR]   org.openrewrite.java.tree.J.accept(J.java:59)
[ERROR]   org.openrewrite.TreeVisitor.visit(TreeVisitor.java:250)
[ERROR]   org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:320)
[ERROR]   org.openrewrite.java.JavaVisitor.visitRightPadded(JavaVisitor.java:1367)
[ERROR]   org.openrewrite.java.JavaVisitor.lambda$visitContainer$35(JavaVisitor.java:1417)
```

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
<!-- Indicate if this is something you would like to work on, and how we can best support you in doing so. -->

--- a/src/main/java/org/openrewrite/java/migrate/joda/ScopeAwareVisitor.java
+++ b/src/main/java/org/openrewrite/java/migrate/joda/ScopeAwareVisitor.java
@@ -34,6 +34,9 @@ class ScopeAwareVisitor extends JavaVisitor<ExecutionContext> {
 
     @Override
     public J preVisit(J j, ExecutionContext ctx) {
+        if (j instanceof J.ClassDeclaration) {
+            scopes.push(new VariablesInScope(getCursor()));
+        }
         if (j instanceof J.Block) {
             scopes.push(new VariablesInScope(getCursor()));
         }
@@ -50,7 +53,13 @@ class ScopeAwareVisitor extends JavaVisitor<ExecutionContext> {
 
     @Override
     public J postVisit(J j, ExecutionContext ctx) {
+        if (j instanceof J.ClassDeclaration) {
+            scopes.pop();
+        }
         if (j instanceof J.Block) {
+            scopes.pop();
+        }
+        if (j instanceof J.MethodDeclaration) {
             scopes.pop();
         }
         return super.postVisit(j, ctx);

--- a/src/test/java/org/openrewrite/java/migrate/joda/JodaTimeRecipeTest.java
+++ b/src/test/java/org/openrewrite/java/migrate/joda/JodaTimeRecipeTest.java
@@ -281,6 +281,45 @@ class JodaTimeRecipeTest implements RewriteTest {
     }
 
     @Test
+    void migrationWithRecord() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.joda.time.DateTime;
+
+              record A(String x) {
+                  public void foo() {
+                      new Bar().bar(new DateTime());
+                  }
+
+                  private static class Bar {
+                      public void bar(DateTime dt) {
+                          dt.getMillis();
+                      }
+                  }
+              }
+              """,
+            """
+              import java.time.ZonedDateTime;
+
+              record A(String x) {
+                  public void foo() {
+                      new Bar().bar(ZonedDateTime.now());
+                  }
+
+                  private static class Bar {
+                      public void bar(ZonedDateTime dt) {
+                          dt.toInstant().toEpochMilli();
+                      }
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
     void migrateMethodWithSafeReturnExpression() {
         //language=java
         rewriteRun(

